engine/dlib/src/dlib/log.cpp

```diff
@@ -507,6 +525,18 @@ void LogInternal(LogSeverity severity, const char* domain, const char* format, .
         return;
     }
 
+    // In release mode, if there are no custom listeners, we'll return here
+    bool is_debug_mode = dLib::IsDebugMode();
+    if (!is_debug_mode && (dmAtomicGet32(&dmLog::g_ListenersCount) == 0))
```

Comment:
TSAN check: We need to check this counter atomically, as it may be changed by another thread.